File: backend/back-pasteleria/src/main/java/cl/milsabores/pasteleria/dto/OrderResponse.java
package cl.milsabores.pasteleria.dto;

import lombok.Builder;
import lombok.Value;

import java.util.List;

@Value
@Builder
public class OrderResponse {
    String id;
    String code;
    Integer subtotal;
    Integer total;
    String couponCode;
    Integer couponDiscount;
    Integer seniorDiscount;
    Integer totalSavings;
    String deliveryOption;
    String address;
    String region;
    String comuna;
    String branchId;
    String branchLabel;
    String pickupDate;
    String pickupTimeSlot;
    String pickupTimeSlotLabel;
    String paymentMethod;
    String paymentMethodLabel;
    String notes;
    String contactEmail;
    String contactPhone;
    String contactName;
    String status;
    String createdAt;
    String updatedAt;
    String statusUpdatedAt;
    String deliveredAt;
    List<OrderItemResponse> items;
    List<StatusHistoryEntry> statusHistory;
    DiscountSummary discounts;

    @Value
    @Builder
    public static class DiscountSummary {
        String couponCode;
        Integer couponDiscount;
        Integer seniorDiscount;
        Integer totalSavings;
    }

    @Value
    @Builder
    public static class StatusHistoryEntry {
        String status;
        String changedAt;
    }
}

File: app/src/main/java/com/example/pasteleriamilssaboresandroid/data/network/dto/OrderResponse.kt
package com.example.pasteleriamilssaboresandroid.data.network.dto

import kotlinx.serialization.Serializable

@Serializable
data class OrderItemResponse(
    val codigo: String,
    val nombre: String,
    val cantidad: Int,
    val precio: Int
)

@Serializable
data class OrderResponse(
    val id: String,
    val code: String,
    val subtotal: Int,
    val total: Int,
    val couponCode: String?,
    val couponDiscount: Int?,
    val seniorDiscount: Int?,
    val totalSavings: Int?,
    val deliveryOption: String?,
    val address: String?,
    val region: String?,
    val comuna: String?,
    val branchId: String?,
    val branchLabel: String?,
    val pickupDate: String?,
    val pickupTimeSlot: String?,
    val pickupTimeSlotLabel: String?,
    val paymentMethod: String?,
    val paymentMethodLabel: String?,
    val notes: String?,
    val contactEmail: String?,
    val contactPhone: String?,
    val contactName: String?,
    val status: String?,
    val createdAt: String?,
    val updatedAt: String?,
    val statusUpdatedAt: String?,
    val deliveredAt: String?,
    val items: List<OrderItemResponse> = emptyList()
)

